exasol/transformers_extension_wrapper.py

```diff
@@ -0,0 +1,239 @@
+                            language_alias: str = LANGUAGE_ALIAS,
+                            run_deploy_container: bool = True,
+                            run_deploy_scripts: bool = True,
+                            run_encapsulate_bfs_credentials: bool = True,
+                            run_encapsulate_hf_token: bool = True) -> None:
+    """
+    Performs all necessary operations to get the Transformers Extension
+    up and running. See the "Getting Started" and "Setup" sections of the
+    extension's User Guide for details.
+
+    Parameters:
+        conf:
+            The secret store. The store should contain all the required
+            parameters for accessing the database and bucket-fs.
+        version:
+            Transformers Extension version. If not specified the hardcoded
+            latest known version will be used.
+        language_alias:
+            The language alias of the extension's language container. Normally
+            this parameter would only be used for testing.
+        run_deploy_container:
+            If set to False will skip the language container deployment.
+        run_deploy_scripts:
+            If set to False will skip the deployment of the UDF scripts.
+        run_encapsulate_bfs_credentials:
+            If set to False will skip the creation of the database connection
+            object encapsulating the bucket-fs credentials.
+        run_encapsulate_hf_token:
+            If set to False will skip the creation of the database connection
+            object encapsulating the Huggingface token.
+    """
+
+    # Make the connection object names
+    bfs_conn_name = "_".join([BFS_CONNECTION_PREFIX, conf.USER])
+    token = conf.get("HF_TOKEN")
+    hf_conn_name = "_".join([HF_CONNECTION_PREFIX, conf.USER]) if token else ""
+
+    if run_deploy_container:
+        deploy_language_container(conf, version, language_alias)
+
+    # Create the required objects in the database
+    if run_deploy_scripts:
+        deploy_scripts(conf, language_alias)
+    if run_encapsulate_bfs_credentials:
+        encapsulate_bucketfs_credentials(
+            conf, path_in_bucket=PATH_IN_BUCKET, connection_name=bfs_conn_name
+        )
+    if token and run_encapsulate_hf_token:
+        encapsulate_huggingface_token(conf, hf_conn_name)
+
+    # Save the connection object names in the secret store.
+    conf.save(BFS_CONNECTION_KEY, bfs_conn_name)
+    conf.save(HF_CONNECTION_KEY, hf_conn_name)
+    # Save the directory names in the secret store
+    conf.save(BFS_MODELS_DIR_KEY, BFS_MODELS_DIR)
+    conf.save(MODELS_CACHE_DIR_KEY, MODELS_CACHE_DIR)
+
+
+def upload_model_from_cache(
+        conf: Secrets,
+        model_name: str,
+        cache_dir: str) -> None:
+    """
+    Uploads model previously downloaded and cached on a local drive. This,
+    for instance, could have been done with the following code.
+
+    from transformers import AutoTokenizer, AutoModel
```

Comment:
Would it maybe make more sense to add another function which combines the from_pretrained and upload